app/models/event.rb

```diff
@@ -119,4 +119,8 @@ def created_at
   def _location_
     @object.location
   end
+
+  def as_json
+    {location: _location_.as_json, agent: agent.as_json, payload: @payload.as_json, created_at: created_at.as_json}
```

Comment:
`agent.as_json` is the full attributes of the Agent? I guess that's okay since we don't have anything sensitive in there, but worth keeping in mind.